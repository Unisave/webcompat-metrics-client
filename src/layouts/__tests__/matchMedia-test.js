/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from "react";
import renderer from "react-test-renderer";

import Component from "..";

jest.mock("react-chartjs-2", () => ({ Bar: () => null }));

it("renders Component matchMedia return true", () => {
  beforeEach(() => {
    fetch.resetMocks();
  });
  fetch.mockResponseOnce(
    JSON.stringify({
      about: "Hourly NeedsDiagnosis issues count",
      date_format: "w3c",
      timeline: [
        {
          count: "822",
          timestamp: "2018-02-08T13:12:16Z",
        },
        {
          count: "819",
          timestamp: "2018-02-08T15:00:00Z",
        },
        {
          count: "806",
          timestamp: "2018-02-09T15:00:03Z",
        },
      ],
    }),
  );
  Object.defineProperty(window, "matchMedia", {
    value: jest.fn(() => {
      return { matches: true };
    }),
  });
  const tree = renderer.create(<Component />).toJSON();
  expect(tree).toMatchSnapshot();
});
