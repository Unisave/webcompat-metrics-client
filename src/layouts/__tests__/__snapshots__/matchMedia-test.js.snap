// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Component matchMedia return true 1`] = `
<div
  className="component"
>
  <header
    className="component"
    role="banner"
  >
    <div
      className="component"
      role="banner"
    >
      <div
        className="menu"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          <span
            className="SVGInline SVGInline--cleaned"
            dangerouslySetInnerHTML={
              Object {
                "__html": "test-file-stub",
              }
            }
          />
        </button>
      </div>
      <a
        href="/"
        onClick={[Function]}
      >
        <div
          className="link"
        >
          <div
            className="baseline"
          >
            <div>
              Webcompat
            </div>
            <div>
              Metrics
            </div>
          </div>
          <div
            className="logo"
          >
            <span
              className="SVGInline SVGInline--cleaned"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "test-file-stub",
                }
              }
            />
          </div>
        </div>
      </a>
    </div>
  </header>
  <div
    className="component "
  >
    <div
      className="container"
    >
      <nav
        className="links"
        role="navigation"
      >
        <a
          aria-current={null}
          href="/needsdiagnosis"
          onClick={[Function]}
        >
          <div
            className="component"
          >
            <div
              className="icon"
            >
              <span
                className="SVGInline SVGInline--cleaned"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "test-file-stub",
                  }
                }
              />
            </div>
            <div
              className="text"
            >
              Needs Diagnosis
            </div>
          </div>
        </a>
        <a
          aria-current={null}
          className={undefined}
          href="/needscontact"
          onClick={[Function]}
          style={undefined}
        >
          <div
            className="component"
          >
            <div
              className="icon"
            >
              <span
                className="SVGInline SVGInline--cleaned"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "test-file-stub",
                  }
                }
              />
            </div>
            <div
              className="text"
            >
              Needs Contact
            </div>
          </div>
        </a>
        <a
          aria-current={null}
          className={undefined}
          href="/needstriage"
          onClick={[Function]}
          style={undefined}
        >
          <div
            className="component"
          >
            <div
              className="icon"
            >
              <span
                className="SVGInline SVGInline--cleaned"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "test-file-stub",
                  }
                }
              />
            </div>
            <div
              className="text"
            >
              Needs Triage
            </div>
          </div>
        </a>
        <a
          aria-current={null}
          href="/weeklyreports"
          onClick={[Function]}
        >
          <div
            className="component"
          >
            <div
              className="icon"
            >
              <span
                className="SVGInline SVGInline--cleaned"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "test-file-stub",
                  }
                }
              />
            </div>
            <div
              className="text"
            >
              Weekly Issue Reports
            </div>
          </div>
        </a>
      </nav>
      <footer
        className="footer"
      >
        <div
          className="socials"
        >
          <a
            className="social"
            href="https://twitter.com/webcompat"
            rel="noopener noreferrer"
            target="_blank"
            title="Twitter webcompat accounts"
          >
            <span
              className="SVGInline SVGInline--cleaned"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "test-file-stub",
                }
              }
            />
          </a>
          <a
            className="social"
            href="https://github.com/webcompat"
            rel="noopener noreferrer"
            target="_blank"
            title="Github webcompat account"
          >
            <span
              className="SVGInline SVGInline--cleaned"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "test-file-stub",
                }
              }
            />
          </a>
        </div>
        <div
          className="footerText"
        >
          <span>
            Created and maintained by webcompat
          </span>
        </div>
      </footer>
    </div>
  </div>
  <main
    className="component isCollapsed"
    role="main"
  >
    <section>
      <div
        className="component"
        style={Object {}}
      >
        <h2
          className="title"
        >
          Needs diagnosis dashboard
        </h2>
        <h3
          className="subtitle"
        >
          Tracking issue diagnosis burndown rate
        </h3>
      </div>
      <header
        className="header"
        style={Object {}}
      >
        <section>
          <h3
            className="headerTitle"
          >
            25 April 2018 - 25 May 2018
          </h3>
        </section>
        <section>
          <form
            onSubmit={[Function]}
          >
            <input
              className="component"
              name="from"
              onChange={[Function]}
              placeholder="From"
              type="date"
              value="2018-04-25"
            />
            <input
              className="component"
              name="to"
              onChange={[Function]}
              placeholder="To"
              type="date"
              value="2018-05-25"
            />
            <button
              className="component"
              onBlur={[Function]}
              onClick={[Function]}
              style={Object {}}
              type="submit"
            >
              Filtered
            </button>
          </form>
        </section>
      </header>
      <article
        className="container"
      >
        <div
          className="wrapper"
        >
          <h3
            className="title"
          >
            Rendering Chart
          </h3>
          <div>
            <div
              className="fetch"
            >
              <span
                className="SVGInline SVGInline--cleaned"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "test-file-stub",
                  }
                }
              />
              <div
                className="fetchTitle"
              >
                Rendering Chart...
              </div>
            </div>
          </div>
        </div>
      </article>
    </section>
  </main>
</div>
`;
